src/bin/ff-analogy.rs

```diff
@@ -0,0 +1,102 @@
+use std::io::BufRead;
+
+use clap::{App, AppSettings, Arg, ArgMatches};
+use finalfusion::similarity::Analogy;
+use finalfusion_utils::{read_embeddings_view, EmbeddingFormat};
+use stdinout::{Input, OrExit};
+
+static DEFAULT_CLAP_SETTINGS: &[AppSettings] = &[
+    AppSettings::DontCollapseArgsInUsage,
+    AppSettings::UnifiedHelpMessage,
+];
+
+fn parse_args() -> ArgMatches<'static> {
+    App::new("ff-analogy")
+        .settings(DEFAULT_CLAP_SETTINGS)
+        .arg(
+            Arg::with_name("format")
+                .short("f")
+                .value_name("FORMAT")
+                .help("Embedding format: finalfusion, finalfusion_mmap, word2vec, text, or textdims (default: finalfusion)")
```

Comment:
This is not yet done in `ff-similar`, but the possible options can be specified with `possible_values`:

https://docs.rs/clap/2.33.0/clap/struct.Arg.html#method.possible_values

The default value with `default_value`:

https://docs.rs/clap/2.33.0/clap/struct.Arg.html#method.default_value